src/DataProtection/DataProtection/src/AuthenticatedEncryption/CngCbcAuthenticatedEncryptorFactory.cs

```diff
@@ -55,8 +55,9 @@ public CngCbcAuthenticatedEncryptorFactory(ILoggerFactory loggerFactory)
             return null;
         }
 
+        using var key = new Secret(secret);
         return new CbcAuthenticatedEncryptor(
-            keyDerivationKey: new Secret(secret),
+            keyDerivationKey: key,
```

Comment:
I believe the secret is disposed here

https://github.com/dotnet/aspnetcore/blob/c8bd3c5eb8e30c70f7b7530307849b29ba417a80/src/DataProtection/DataProtection/src/Cng/CbcAuthenticatedEncryptor.cs#L43

https://github.com/dotnet/aspnetcore/blob/c8bd3c5eb8e30c70f7b7530307849b29ba417a80/src/DataProtection/DataProtection/src/Cng/CbcAuthenticatedEncryptor.cs#L207

Also I think the changed that you have maid wouldn't work correctly. You are disposing the `Secret` instance and now the `CbcAuthenticatedEncryptor` instance that is returned has a disposed secret which would not work.

Reply:
created provider (ISP800_108_CTR_HMACSHA512Provider ) doesn't store Secret instance, it stores it as plain text:

https://github.com/dotnet/aspnetcore/blob/main/src/DataProtection/DataProtection/src/SP800_108/SP800_108_CTR_HMACSHA512Util.cs#L53

Reply:
Oh I see! I didn't know that. 
How about disposing the `Secret` in the `CbcAuthenticatedEncryptor.Dispose` method? Would that work too?

Reply:
It's a good proposal

CbcAuthenticatedEncryptor          doesnt' store Secret (uses ISP800_108_CTR_HMACSHA512Provider)
CngGcmAuthenticatedEncryptor  doesn't store Secret (uses ISP800_108_CTR_HMACSHA512Provider)
AesGcmAuthenticatedEncryptor   copies Secret and dipose copy (not instance passed in contructor)
ManagedAuthenticatedEncryptor stores passed Secret and dispose it

I can do everything uniformly

Reply:
I have reviewed the DataProtection code.

I see two solutions if we make the behaviour for encryptors identical

1. Secret managed inside AuthEncryptors
	- need to change 3 encryptors 
		- CbcAuthenticatedEncryptor
		- CngGcmAuthenticatedEncryptor
		- AesGcmAuthenticatedEncryptor 
	- CbcAuthenticatedEncryptor and CngGcmAuthenticatedEncryptor don't need to store the secret, because it use ISP800_108_CTR_HMACSHA512Provider which copy value from Secret.
	
2. Secret managed outside AuthEncryptors
	- need to change 1 encryptor (ManagedAuthenticatedEncryptor)

If we don't need identical behaviour I can dispose passed Secret in *Encryptor.Dispose() and it will works


@Kahbazi , which approach do you think is preferable?